hivemq-edge/src/frontend/src/extensions/datahub/designer/data_policy/DataPolicyNode.utils.ts

```diff
@@ -0,0 +1,116 @@
+import { getIncomers, getOutgoers, Node } from 'reactflow'
+
+import { DataPolicy, DataPolicyValidator, PolicyOperation } from '@/api/__generated__'
+import { DataHubNodeType, DataPolicyData, DryRunResults, WorkspaceState } from '@datahub/types.ts'
+import { PolicyCheckErrors } from '@datahub/designer/validation.errors.ts'
+import { isTopicFilterNodeType } from '@datahub/utils/node.utils.ts'
+
+/* istanbul ignore next -- @preserve */
+export function getSubFlow(source: Node, acc: Node[], store: WorkspaceState, only = false) {
+  const allIds = Array.from(new Set(acc.map((e) => e.id)))
+  const { nodes, edges } = store
+
+  // TODO[NVL] This is wrong: should check it's from the right handles
+  const incomers = getIncomers(source, nodes, edges).filter(
+    (node) =>
+      !allIds.includes(node.id) &&
+      node.type !== DataHubNodeType.DATA_POLICY &&
+      node.type !== DataHubNodeType.BEHAVIOR_POLICY
+  )
+  const outgoers = getOutgoers(source, nodes, edges).filter(
+    (node) =>
+      !allIds.includes(node.id) &&
+      node.type !== DataHubNodeType.DATA_POLICY &&
+      node.type !== DataHubNodeType.BEHAVIOR_POLICY
+  )
+
+  acc.push(...incomers, ...outgoers)
+
+  if (outgoers.length) {
+    outgoers.forEach((node) => {
+      const subFlow = getSubFlow(node, acc, store)
+      acc = Array.from(new Set([...acc, ...subFlow]))
+    })
+  }
+
+  if (incomers.length && !only) {
+    incomers.forEach((node) => {
+      const subFlow = getSubFlow(node, acc, store)
+      acc = Array.from(new Set([...acc, ...subFlow]))
```

Comment:
question(non-blocking): I see here multiple remapping happening with sets. `Array.from(new Set([...])` I do not see it as concerning but maybe we can reduce the amount of iterations by utilising lodash's `uniq` function?
https://lodash.com/docs/4.17.15#uniq

WDYT?

Reply:
That's a funny one. If i'm not using lodash already, then I skip it totally. I prefer to check individual alternatives. 
The one of `uniq` is suggesting using Set: https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore?tab=readme-ov-file#_uniq